ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -431,6 +435,24 @@ public void shutdown() throws StoreException {
   }
 
   /**
+   * @return the {@link DiskSpaceRequirements} for this store to provide to
+   * {@link DiskSpaceAllocator#initializePool(Collection)}. This will be {@code null} if this store uses a non-segmented
+   * log. This is because it does not require any additional/swap segments.
+   * @throws StoreException
+   */
+  DiskSpaceRequirements getDiskSpaceRequirements() throws StoreException {
+    synchronized (lock) {
+      checkStarted();
+      DiskSpaceRequirements requirements = log.getDiskSpaceRequirements();
```

Comment:
would recommend exposing a method in log to get the unallocated segments count. You can then generate new DiskSpaceRequirements(log.getSegmentCapacity(), log.getUnAllocatedSegmentsCount(), compactor.getSwapSegmentsInUse()). This gives a notion that log might have some swap segments on its own apart from compaction